src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.ts

```diff
@@ -145,6 +146,11 @@ export class RbdTrashListComponent implements OnInit {
       }
       viewCacheStatusMap[pool.status].push(pool.pool_name);
       images = images.concat(pool.value);
+      if(images.length > 0){
+        this.disablePurgeBtn = true;
+      } else {
+        this.disablePurgeBtn = false;
+      }
```

Comment:
Mmmm, I don't understand this: when no images, `disablePurgeBtn` is `false`, but then it's 'negated' in `[disabled]="!disablePurgeBtn"`, so eventually it behaves like `true`. Am I missing something?

Reply:
You are right, I would like to see that more straight forward, by writing `[disabled]="images.length === 0"` in the template. `images` will be set as `this.images` in the next step (see line 132).

Reply:
I agree: even better encapsulate
```
images && images.length === 0
```
in a method : ```[disabled]="disablePurgeTrashButton()"```

Reply:
Please be careful with methods in templates. Angular has no way to check whether a method return value has changed unless it runs it (contrary to class members or pure pipes), so it leads to running methods 12 times (1 per lifecycle hook) per change detection cycle. Unless we start moving our components to OnPush Change Detection Strategy or we [use pure methods with this recipe](https://medium.com/angular-in-depth/tiny-angular-pipe-to-make-any-function-memoizable-f6c8fa917f2f), we should avoid as much as possible methods in templates (with a x12 amplification factor even accessors functions are no longer negligible).

Reply:
> You are right, I would like to see that more straight forward, by writing `[disabled]="images.length === 0"` in the template. `images` will be set as `this.images` in the next step (see line 132).
Here images are coming undefined

Reply:
> Please be careful with methods in templates. Angular has no way to check whether a method return value has changed unless it runs it (contrary to class members or pure pipes), so it leads to running methods 12 times (1 per lifecycle hook) per change detection cycle. Unless we start moving our components to OnPush Change Detection Strategy or we [use pure methods with this recipe](https://medium.com/angular-in-depth/tiny-angular-pipe-to-make-any-function-memoizable-f6c8fa917f2f), we should avoid as much as possible methods in templates (with a x12 amplification factor even accessors functions are no longer negligible).

Right, I missed this in my suggestion (even ChangeDetectionStrategy.OnPush don't help much in these cases).

Reply:
`OnPush` at least reduces the amount of events that trigger change detection and hence minimizes how often template expressions are evaluated.

Reply:
> `OnPush` at least reduces the amount of events that trigger change detection and hence minimizes how often template expressions are evaluated.

I know, what I meant is ```onPush``` does not help when change detection is triggered from within the component itself.